### PR TITLE
Refs task #387246 Unit test update payload

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets

- [#387246](https://dev.sun-asterisk.com/issues/387246)

## What's this PR do ?

- [x] Unit test for payload controller (update)
- [x] Unit test for payload repository (update)

## Library
*(List package, library third party add new include version)*

- N/A

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [x] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [x] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
*(Other notes)*
image

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr></table>

File: tests/Feature/PayloadControllerTest.php [hunk 1]
```diff
@@ -170,4 +171,149 @@ public function testStorePayloadOfWebhookNotExistFeature()
...
+    {
+        $user = factory(User::class)->create();
+        $webhook = factory(Webhook::class)->create(['user_id' => $user->id]);
+        $this->actingAs($user);
+        $response = $this->put(route('webhooks.payloads.update', ['webhook' => $webhook, 'payload' => -1]), [
+            'content' => 'New content'
+        ]);
+
+        $response->assertStatus(404);
+    }
```
Comment (tamtt-1031) [inline tests/Feature/PayloadControllerTest.php:388]: đợi pull của a merged rồi bổ sung case phân quyền vào luôn nghe :D

**Review by ghost — APPROVED** [review]

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr></table>

File: tests/Feature/PayloadControllerTest.php (hunk 1 above)
Comment (ghost) [reply tests/Feature/PayloadControllerTest.php:388]: em bổ sung ở [đây](https://github.com/sun-asterisk-research/chatwork-forwarder/pull/46/files#diff-8b7ba6662bb27b0f874dedc382914a0eR261) và [đây](https://github.com/sun-asterisk-research/chatwork-forwarder/pull/46/files#diff-8b7ba6662bb27b0f874dedc382914a0eR374) rồi ạ

**Review by nguyenthinhhn — APPROVED** [review]

File: tests/Feature/PayloadControllerTest.php [hunk 2]
```diff
@@ -206,4 +207,185 @@ public function testStorePayloadOfWebhookNotExistFeature()
...
+
+    /**
+     * test Feature show payload not belong to webhook
+     *
+     * @return void
+     */
+    public function testShowPayloadNotBelongToWebhookFeature()
```
Comment (tamtt-1031) [inline tests/Feature/PayloadControllerTest.php:265]: còn 1 case payload thuộc về webhook ko phải của current user nửa e

Comment (ghost) [reply tests/Feature/PayloadControllerTest.php:265]: em thêm rồi [đây](https://github.com/sun-asterisk-research/chatwork-forwarder/pull/46/files#diff-8b7ba6662bb27b0f874dedc382914a0eR278) và [đây](https://github.com/sun-asterisk-research/chatwork-forwarder/pull/46/files#diff-8b7ba6662bb27b0f874dedc382914a0eR409)

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr></table>

**Review by tamtt-1031 — APPROVED** [review]

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**